packages/neon_framework/packages/account_repository/test/account_repository_test.dart

```diff
@@ -0,0 +1,564 @@
+                  'biographyScope': 'v2-local',
+                  'displaynameScope': 'v2-federated',
+                  'email': 'admin@example.com',
+                  'emailScope': 'v2-federated',
+                  'enabled': true,
+                  'fediverseScope': 'v2-local',
+                  'headlineScope': 'v2-local',
+                  'organisationScope': 'v2-local',
+                  'phoneScope': 'v2-local',
+                  'profile_enabledScope': 'v2-local',
+                  'roleScope': 'v2-local',
+                  'storageLocation': '/usr/src/nextcloud/data/admin',
+                  'twitterScope': 'v2-local',
+                  'websiteScope': 'v2-local',
+                },
+              },
+            }),
+            null,
+          ),
+        );
+
+        expect(
+          repository.getAccount(accountList.first),
+          completion(isA<Account>().having((e) => e.username, 'username', equals('admin'))),
+        );
+      });
+
+      test('rethrows http exceptions as `FetchAccountFailure`', () async {
+        when(() => users.getCurrentUser()).thenThrow(http.ClientException(''));
+
+        await expectLater(
+          repository.getAccount(accountList.first),
+          throwsA(isA<FetchAccountFailure>().having((e) => e.error, 'error', isA<http.ClientException>())),
+        );
+      });
+    });
+
+    group('logIn', () {
+      setUp(() async {
+        when(() => storage.saveAccounts(any())).thenAnswer((_) async => {});
+        when(() => storage.saveLastAccount(any())).thenAnswer((_) async => {});
+      });
+
+      final account = Account((b) {
+        b
+          ..credentials.replace(accountList.first)
+          ..client = _FakeNextcloudClient();
+      });
+
+      test('adds account and sets active if null', () async {
+        await repository.logIn(account);
+
+        expect(
+          repository.accounts,
+          emits(
+            isA<_AccountStream>()
+                .having((e) => e.accounts, 'accounts', hasLength(1))
+                .having((e) => e.active, 'active', equals('b0682d652840ef50a4115cc77109bedd8c577ccc')),
+          ),
+        );
+        verify(() => storage.saveAccounts(any(that: contains(accountList.first)))).called(1);
+        verify(() => storage.saveLastAccount(any(that: equals('b0682d652840ef50a4115cc77109bedd8c577ccc')))).called(1);
+      });
+    });
+
+    group('logOut', () {
+      setUp(() async {
+        when(() => storage.readAccounts()).thenAnswer((_) async => accountList);
+        when(() => storage.saveLastAccount(any())).thenAnswer((_) async => {});
+        when(() => storage.saveAccounts(any())).thenAnswer((_) async => {});
+
+        await repository.loadAccounts();
+      });
+
+      test('removes active account', () async {
+        when(() => appPassword.deleteAppPassword()).thenAnswer(
+          (_) async => _DynamiteResponseMock<core.AppPasswordDeleteAppPasswordResponseApplicationJson, void>(),
+        );
```

Comment:
Verify that it was called once

Reply:
This is the setup part of the logout code.
1. it is a test setup so it should expect or validate anything
2. it is the logout flow so it should test the account loading

Reply:
I don't understand what you mean :thinking:

Reply:
My bad. The github UI displayed the comment in the wrong place after force pushing so I got confused